docs/rfc/009-bit-byte-conversions.md

```diff
@@ -8,237 +8,330 @@ The Aleo Team.
-function u64_to_bytes_le(x: u64) -> [u8; 8];
-function u64_to_bytes_be(x: u64) -> [u8; 8];
-function u128_to_bytes_le(x: u128) -> [u8; 16];
-function u128_to_bytes_be(x: u128) -> [u8; 16];
-
-// signed to bytes, little and big endian
-function i16_to_bytes_le(x: i16) -> [u8; 2];
-function i16_to_bytes_be(x: i16) -> [u8; 2];
-function i32_to_bytes_le(x: i32) -> [u8; 4];
-function i32_to_bytes_be(x: i32) -> [u8; 4];
-function i64_to_bytes_le(x: i64) -> [u8; 8];
-function i64_to_bytes_be(x: i64) -> [u8; 8];
-function i128_to_bytes_le(x: i128) -> [u8; 16];
-function i128_to_bytes_be(x: i128) -> [u8; 16];
-
-// unsigned from bytes, little and big endian
-function u16_from_bytes_le(x: [u8; 2]) -> u16;
-function u16_from_bytes_be(x: [u8; 2]) -> u16;
-function u32_from_bytes_le(x: [u8; 4]) -> u32;
-function u32_from_bytes_be(x: [u8; 4]) -> u32;
-function u64_from_bytes_le(x: [u8; 8]) -> u64;
-function u64_from_bytes_be(x: [u8; 8]) -> u64;
-function u128_from_bytes_le(x: [u8; 16]) -> u128;
-function u128_from_bytes_be(x: [u8; 16]) -> u128;
-
-// signed from bytes, little and big endian
-function i16_from_bytes_le(x: [u8; 2]) -> i16;
-function i16_from_bytes_be(x: [u8; 2]) -> i16;
-function i32_from_bytes_le(x: [u8; 4]) -> i32;
-function i32_from_bytes_be(x: [u8; 4]) -> i32;
-function i64_from_bytes_le(x: [u8; 8]) -> i64;
-function i64_from_bytes_be(x: [u8; 8]) -> i64;
-function i128_from_bytes_le(x: [u8; 16]) -> i128;
-function i128_from_bytes_be(x: [u8; 16]) -> i128;
+// unsigned:
+let x: u32 = 10;
+let xle: [u8; 4] = x.to_bytes_le();
+let xbe: [u8; 4] = x.to_bytes_be();
+console.assert(xle == [10, 0, 0, 0]);
+console.assert(xbe == [0, 0, 0, 10]);
+console.assert(x == u8::from_bytes_le(xle));
+console.assert(x == u8::from_bytes_be(xbe));
+// signed:
+let y: i32 = -2147483648;
+let yle: [u8; 4] = y.to_bytes_le();
+let ybe: [u8; 4] = y.to_bytes_be();
+console.assert(yle == [0, 0, 0, 128]);
+console.assert(ybe == [128, 0, 0, 0]);
+console.assert(y == u8::from_bytes_le(yle));
+console.assert(y == u8::from_bytes_be(ybe));
 ```
 
-## Handling of the Native Functions
+Conversions between `u8` and bytes amount to
+converting between the integer and the singleton array that contains the integer;
+in this case, there is no difference between little and big endian order.
+These conversions are available for completeness, but may not be very useful in practice.
+
+Conversions between `i8` and bytes amount to
+converting between the integer and the singleton array that contains the integer re-interpreted as unsigned;
+in this case, there is no different between little and big endian order.
+Thus, these conversions may be used to re-interpret between signed and unsigned bytes,
+but arguably there should be more dedicated and comprehensive operations for that in Leo:
+that is an independent extension of Leo,
+which may make the conversions between `i8` and bytes just available for completeness, but not very useful in practice.
```

Comment:
The sentence ending on this line is long and hard to parse.

Reply:
I just pushed some edits that attempt to clarify. If that seems fine, you can resolve this conversation.